File: AuthenticationSOAP/AuthenticationSOAP/OperationResultValue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace AuthenticationSOAP
{
    public class OperationResultValue<T> : OperationResult
    {
        public T ResultValue;

        public OperationResultValue(T resultValue)
            : base()
        {
            this.ResultValue = resultValue;
        }
        public OperationResultValue()
            : base(ErrorEnum.ResultNotInitialized, "Result not initialized")
        {
        }
        public OperationResultValue(ErrorEnum error, string errorString)
            : base(error, errorString)
        {
        }
    }
}

File: LibraryBooksOfInterest/src/book/of/interest/om/OperationResultValue.java
package book.of.interest.om;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlRootElement;
import javax.xml.bind.annotation.XmlSeeAlso;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
@XmlSeeAlso({BookOfInterest.class, Location.class, User.class, OperationResult.class})
public class OperationResultValue<T> extends OperationResult{

    public T ResultValue;

    public OperationResultValue()
    {
    	super(ErrorEnum.ResultNotInitialized, "Result not initialized");
    }

    public OperationResultValue(T resultValue)
    {
        this.ResultValue = resultValue;
    }

    public OperationResultValue(ErrorEnum error, String errorString)
    {
    	super(error, errorString);
    }
}
